### ISSUE TITLE
Type naming collisions are silent and unexpected

### ISSUE THREAD

**Issue by RoarkeRandall** [graphql-dotnet/graphql-dotnet#1698]
## Description

It is very easy to have multiple types that end up causing naming collisions. It will compile & run, but somewhat randomly select one of the types with the naming collision. 

### Steps to reproduce

Create some type's:

```
    public class TemplateType : ObjectGraphType<Content.Template>
    {
        public TemplateType()
        {
            Field<TemplateTypeEnum>(
                "templateType",
                "The template's TemplateType",
                resolve: ctx => ctx.Source.TemplateType
            );
        }
    }

    public class TemplateTypeEnum : EnumerationGraphType<Content.TemplateType>
    { }
```

Reference `TemplateType` in other types:

```
        public RenderedTemplateType()
        {
            Field<TemplateType>(
                "template",
                "The template that was rendered",
                resolve: ctx => ctx.Source.Template
            );
        }
```
### Expected result

`RenderedTemplateType` should have a member `template` that is of type `TemplateType`

### Actual result

Everything builds & runs, but `RenderTemplateType` ends up having a member `template` of type `TemplateTypeEnum`. I assume this is because `TemplateTypeEnum` wraps type `Content.TemplateType`

### Environment

version 2.4.0
dotnet core 2.2
macOS 10.15.4

**Comment by RoarkeRandall** [comment]
Renaming `TemplateType` to `Template` is a workaround for this. It also fixes this issue: https://github.com/graphql-dotnet/graphql-dotnet/issues/1692

**Comment by sungam3r** [comment]
https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Types/Scalars/EnumerationGraphType.cs#L82 . Note -- where constraint was added.

A lot of naming issues was fixed in 3.0.0-preview. Please, try it.
